src/ert/parsing/lark_parser.py

```diff
@@ -210,9 +209,19 @@ def _tree_to_dict(
             args = _substitute_args(args, constraints, defines)
             value_list = constraints.apply_constraints(args, kw, cwd)
 
-            if constraints.multi_occurrence:
-                arglist = config_dict.get(kw, [])
-                arglist.append(value_list)  # type: ignore
+            arglist = config_dict.get(kw, [])
+            if kw == "DEFINE":
+                define_key, *define_args = value_list
+                existing_define = next(
```

Comment:
So there were some questions regarding whether this is expected behavior regarding situations where order is significant. Order should never be significant given that the user writes `define_key` as "<ARG>" which I believe we require now. However, if the `define_key` for instance was `<A<B><C>>` and you had other dines such as `<AB<C>>` and `<A<B>C>` then order would be significant. I am wondering whether we should write this as

`define = [kw, arg for kw, arg in arglist if kw != define_key] + [(define_key, define_args)]` to have a consistent order of defines.